### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by erickma1**
In this milestone, we added unit tests for all the methods in all the classes namely:

- [ ] Book
- [ ] Person
- [ ] Student
- [ ] Teacher
- [ ] Classroom
- [ ] Rental
- [ ] all classes that were created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    it 'should be an instance of Person' do
+      expect(person).to be_an_instance_of(Person)
+    end
+
+    it 'should have default attributes' do
+      expect(person.name).to eql('John Doe')
+      expect(person.age).to eql(25)
+      expect(person.parent_permission).to be true
+    end
+  end
```
Comment (cynthiainga) [inline spec/person_spec.rb]: - [ ] You should test the scenario of when the name is not given it should return the default value which is `Unknown` for the name 🙃 The constructor method sets the default value for the optional name parameter if it is not provided.
- [ ] Also the constructor method should return a Person object with provided empty rentals

File: spec/student_spec.rb
Comment (cynthiainga) [inline spec/student_spec.rb:1]: - [ ] As mentioned in the project requirements all the methods in all the classes should be tested, kindly test the `play_hooky` method 😁 😁

File: spec/capitalizedecorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative '../capitalizedecorator'
+
+describe CapitalizeDecorator do
+  before :each do
+    @capitalize_decorator = CapitalizeDecorator.new('Ichsan')
+  end
+
+  it 'should return an instance of CapitalizeDecorator' do
+    expect(@capitalize_decorator).to be_an_instance_of CapitalizeDecorator
+  end
```
Comment (cynthiainga) [inline spec/capitalizedecorator_spec.rb]: - [ ] Here you are supposed to test if the Capitalize method works, for example, if you have `ichsan` as name it is supposed to correct the name by using the `correct_name` method and should be eq `Ichsan` 🙃

File: spec/trimmerdecorator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require_relative '../trimmerdecorator'
+
+describe TrimmerDecorator do
+  before :each do
+    @trimmer_decorator = TrimmerDecorator.new('Ichsan')
+  end
+
+  it 'should return an instance of TrimmerDecorator' do
+    expect(@trimmer_decorator).to be_an_instance_of TrimmerDecorator
+  end
```
Comment (cynthiainga) [inline spec/trimmerdecorator_spec.rb]: - [ ] Same remark, write a test to show that the trimmer_decorator works, for example, if the name is `Jonathanuous` it should trimmer and correct the name by using the `correct_name` method and supposed to be eq `Jonathanuo` 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @erickma1, @HenaryKhuraijam & @Nuga0909, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] All test files are placed in a different directory `(/spec)` ✔️ 
- [ ] There is one file per class ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
- Wrote code to test the scenario of when the name is not given it should return the default value which is Unknown for the name, the constructor method sets the default value for the optional name parameter if it is not provided.
- Test the play_hooky method 
- Test if the Capitalize method works, for example, if you have ichsan as name it is supposed to correct the name by using the correct_name method and should be eq Ichsan
- Modified trimmer_decorator so that for example, if the name is Jonathanuous it would trim and correct the name by using the correct_name method and supposed to be eq Jonathanuo

**Review by Gambit142 — APPROVED** [review]
Hi @erickma1 @HenaryKhuraijam @Nuga0909 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Use of Rspec testing library ✔️ 
- Wrote unit tests for all methods of required classes ✔️ 
- Tested attributes of classes ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**